### PR TITLE
Supporting keyword and exact phrase searching

### PR DESCRIPTION
**Pull request by bistline**
Previously, searching only supported keyword or a single phrase when performing searches.  We will now be able to do multiple phrases, or a combination of phrase and keyword.  Also, study matches will report which terms matched during search (before it just returned all search terms).

This PR satisfies SCP-2217.

### CODE REVIEW

File: app/controllers/api/v1/search_controller.rb [hunk 1]
```diff
@@ -525,6 +536,21 @@ def sanitize_search_values(terms)
         end
       end
 
+      # extract any "quoted phrases" from query string and tokenize terms
+      def self.extract_phrases_from_search(query_string:)
+        terms = []
+        query_string.split("\"").each do |substring|
+          # when splitting on double quotes, phrases will not have any leading/trailing whitespace
```
Comment (devonbush) [inline app/controllers/api/v1/search_controller.rb]: do we need to escape regex special characters that are inside terms?

Comment (bistline) [reply app/controllers/api/v1/search_controller.rb]: That's already happened in https://github.com/broadinstitute/single_cell_portal_core/blob/jb-keyword-phrase-search/app/controllers/api/v1/search_controller.rb#L147 - `sanitize_search_terms` escapes any unsafe characters.

Comment (bistline) [reply app/controllers/api/v1/search_controller.rb]: I actually just realized that this isn't true in this case - it's removing some unsafe characters, but not ones specific to regular expressions.  We want to use `Regexp.quote` for that.  I'll make that change.

Comment (bistline) [reply app/controllers/api/v1/search_controller.rb]: c9a63d2

**Pull request merged**